src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -169,6 +169,22 @@ FORBID_C_FUNCTION(char* strtok(char*, const char*), "use strtok_r");
 FORBID_C_FUNCTION(int vsprintf(char*, const char*, va_list), "use os::vsnprintf");
 FORBID_C_FUNCTION(int vsnprintf(char*, size_t, const char*, va_list), "use os::vsnprintf");
 
+// All of the following functions return raw C-heap pointers (sometimes as an option, e.g. realpath or getwd)
+// or, in case of free(), take raw C-heap pointers. Don't use them unless you are really sure you must.
+FORBID_C_FUNCTION(void* malloc(size_t size), "use os::malloc");
+FORBID_C_FUNCTION(void* calloc(size_t nmemb, size_t size), "use os::malloc and zero out manually");
+FORBID_C_FUNCTION(void free(void *ptr), "use os::free");
+FORBID_C_FUNCTION(void* realloc(void *ptr, size_t size), "use os::realloc");
+FORBID_C_FUNCTION(char* strdup(const char *s), "use os::realloc");
+FORBID_C_FUNCTION(char* strndup(const char *s, size_t n), "use os::strdup");
```

Comment:
I take it there are no calls to `::strndup`?  If there were, we should probably add `os::strndup`.

Reply:
There is one case, in share/compiler/directivesParser.cpp, but commented out.